xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -131,23 +137,129 @@ XdsClient createXdsClient() {
     };
     xdsClientPool = new RefCountedXdsClientObjectPool(xdsClientFactory);
     xdsClient = xdsClientPool.getObject();
-    xdsClient.watchConfigData(authority, new ConfigWatcher() {
-      @Override
-      public void onConfigChanged(ConfigUpdate update) {
+    xdsClient.watchConfigData(authority, new ConfigWatcherImpl(listener));
+  }
+
+  private class ConfigWatcherImpl implements ConfigWatcher {
+
+    final Listener2 listener;
+
+    ConfigWatcherImpl(Listener2 listener) {
+      this.listener = listener;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void onConfigChanged(ConfigUpdate update) {
+      Map<String, ?> config;
+      if (update.getRoutes().size() > 1) {
```

Comment:
The conditions of this if-elseif-else is verbose and unclear. Use `null` value of fields in `ConfigUpdate` to make this cleaner:

```java
if (update.clusterName != null) {
   // CDS
} else if (update.getRoutes.size() == 1) {
   // weighted-target
} else {
   // xds-routing
}
```